File: app/src/main/java/com/dumuzeyn/mp3player/VoltuneLog.java
package com.dumuzeyn.mp3player;

import android.util.Log;

/** Debug-only logging that never records URIs, paths, titles, or exception messages. */
public final class VoltuneLog {
    private static final String TAG = "VoltuneDebug";

    private VoltuneLog() {
    }

    public static void info(String event) {
        if (BuildConfig.DEBUG) {
            Log.i(TAG, event);
        }
    }

    public static void warning(String event) {
        if (BuildConfig.DEBUG) {
            Log.w(TAG, event);
        }
    }

    public static void failure(String event, Throwable error) {
        if (BuildConfig.DEBUG) {
            String category = error == null ? "unknown" : error.getClass().getSimpleName();
            Log.e(TAG, event + " category=" + category);
        }
    }
}

File: app/src/main/java/com/dumuzeyn/mp3player/VoltuneLog.kt
package com.dumuzeyn.mp3player

import android.util.Log

/** Debug-only logging that never records URIs, paths, titles, or exception messages. */
object VoltuneLog {
    private const val TAG = "VoltuneDebug"

    @JvmStatic
    fun info(event: String) {
        if (BuildConfig.DEBUG) Log.i(TAG, event)
    }

    @JvmStatic
    fun warning(event: String) {
        if (BuildConfig.DEBUG) Log.w(TAG, event)
    }

    @JvmStatic
    fun failure(event: String, error: Throwable?) {
        if (BuildConfig.DEBUG) {
            val category = error?.javaClass?.simpleName ?: "unknown"
            Log.e(TAG, "$event category=$category")
        }
    }
}
